### PR TITLE
Servicenow bug fix

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/28820

## Description
Fixed mirror closing ticket/incident for ticket type of `sc_task`

## Minimum version of Demisto
- [x] 6.0.0

## Does it break backward compatibility?
   - [x] No

### CODE REVIEW

**Comment by ShirleyDenkberg** [comment]
@Itay4 Doc review completed.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 1]
```diff
@@ -2034,6 +2034,8 @@ def update_remote_system_command(client: Client, args: Dict[str, Any], params: D
     ticket_id = parsed_args.remote_incident_id
     if parsed_args.incident_changed:
         demisto.debug(f'Incident changed: {parsed_args.incident_changed}')
+        if ticket_type == 'sc_task' and parsed_args.inc_status == 2 and params.get('close_ticket'):
```
Comment (Itay4) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:2037]:
```suggestion
        if ticket_type == 'sc_task' and parsed_args.inc_status == IncidentStatus.DONE and params.get('close_ticket'):
```

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 2]
```diff
@@ -2034,6 +2034,8 @@ def update_remote_system_command(client: Client, args: Dict[str, Any], params: D
...
+            parsed_args.data['state'] = '3'
```
Comment (Itay4) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:2038]: please add comment here why is this done

**Review by Itay4 — CHANGES_REQUESTED** [review]
please add test

**Comment by Shellyber** [comment]
Tested it also in the customer env, works great.

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py [hunk 3]
```diff
@@ -1994,7 +1994,7 @@ def get_remote_data_command(client: Client, args: Dict[str, Any], params: Dict)
         user_email = user.get('email')
         ticket['caller_id'] = user_email
 
-    if ticket.get('resolved_by'):
+    if ticket.get('resolved_by') or ticket.get('closed_at'):
```
Comment (Itay4) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2.py:1997]: is there test for `closed_at` ?

File: Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py [hunk 4]
```diff
@@ -583,3 +583,41 @@ def test_upload_entries_update_remote_system_command(mocker):
     mocker.patch.object(client, 'add_comment', side_effect=add_comment_request)
 
     update_remote_system_command(client, args, params)
+
+
+DATA = {'close_notes': 'This is closed', 'closed_at': '2020-10-29T13:19:07.345995+02:00', 'impact': '3',
+        'priority': '4', 'resolved_at': '2020-10-29T13:19:07.345995+02:00', 'severity': '1 - Low',
+        'short_description': 'Post parcel', 'sla_due': '0001-01-01T00:00:00Z', 'urgency': '3', 'state': '3',
```
Comment (Itay4) [inline Packs/ServiceNow/Integrations/ServiceNowv2/ServiceNowv2_test.py:590]: remove from here all the fields that aren't actually sent from the server, such as `state`
change `DATA` to an indicative name

**Review by Itay4 — APPROVED** [review]

**Pull request merged**